src/marching_cubes.jl

```diff
@@ -3,41 +3,26 @@
 include("lut/mc.jl")
 
 """
-`marching_cubes(sdf::SignedDistanceField, [iso = 0.0,] [MT = HomogenousMesh{Point{3,Float64},Face{3,Int}}])`
-
 Construct a `HomogenousMesh` from a `SignedDistanceField` using the
 marching cubes algorithm. This method is faster than Marching Tetrahedra
 and generates fewer vertices and faces (about 1/4 as many).
 However it may generate non-manifold meshes, while Marching
 Tetrahedra guarentees a manifold mesh.
 """
-function marching_cubes(sdf::SignedDistanceField{3,ST,FT},
-                        iso=0.0,
-                        MT::Type{M}=SimpleMesh{Point{3,Float64},Face{3,Int}},
-                        eps=0.00001, reduceverts=true) where {ST,FT,M<:AbstractMesh}
-    VertType, FaceType = _determine_types(M, FT)
-    marching_cubes(sdf, VertType, FaceType, iso, M, eps, reduceverts)
-end
-
-function marching_cubes(sdf::SignedDistanceField{3,ST,FT}, ::Type{VertType}, ::Type{FaceType},
-                               iso=0.0,
-                               MT::Type{M}=SimpleMesh{Point{3,Float64},Face{3,Int}},
-                               eps=0.00001, reduceverts=true, insidepositive=Val(false)) where {ST,FT,M<:AbstractMesh, VertType, FaceType}
+function isosurface(sdf::AbstractArray{T, 3}, method::MarchingCubes, ::Type{VertType}=SVector{3,Float64}, ::Type{FaceType}=SVector{3, Int};
```

Comment:
@rdeits do you think we should ensure origin and widths are the same as `VertType` for all `isosurface` signatures?